src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -28,7 +28,7 @@ public class RestoreCommand
 
         private readonly RestoreRequest _request;
 
-        private bool _success = true;
+        private bool _success;
```

Comment:
I'm adding it here, but it's a general question about the restore execution.
It's a bit hard to understand all the scenarios given where the additional check is happening. 

Can you explain the following scenario and ensure that it works as expected. 

Say I have 2 package references and both of them have a bad version.
Because we probably fail fast (do we?), the error message talks about the first version. 
Now I fix that one, nomination happens again, and I get another error. is the error message updated? 

As is, I'm afraid that it might no-op and give the old warning message.

Reply:
With the current implementation, no-op is skipped (see the change to `EvaluateCacheFile`), so as long as there are messages in the `RestoreArgs`, it will never no-op. This lets the error message(s) be written to the assets file when the message(s) change.

You're right about it only showing one error at a time at the moment. That's an improvement that can be made in `NominateProjectAsync`, and why the `AdditionalMessages` property in `RestoreArgs` is a list, not a single item.

During development, I had the problem that you described, when the error changed, the assets file wasn't being updated. But it was since fixed, unfortunately by disabling no-op.

Reply:
It's fine if no-op is skipped, it's just a bit difficult to follow. 
Maybe a refactoring is required in that part of the code.
Feel like with noop + lock file + CPVM scenarios things are declared all over the place which made it difficult to read.